modules/service/high-availability-service.js

```diff
@@ -0,0 +1,328 @@
+                this.config.ssl_keypath,
+            ), masterNodeData.kademliaKey);
+        }
+        if (masterNodeData.bootstraps) {
+            fs.writeFileSync(path.join(
+                this.config.appDataPath,
+                'bootstraps.json',
+            ), masterNodeData.bootstraps);
+        }
+        if (masterNodeData.routingTable) {
+            fs.writeFileSync(path.join(
+                this.config.appDataPath,
+                'router.json',
+            ), masterNodeData.routingTable);
+        }
+        this.logger.trace('Synchronizing with master node completed.');
+    }
+
+    startPostgresReplication(remoteHostname, remoteOpDbUsername) {
+        this.logger.trace('Starting postgres replication');
+        execSync('/etc/init.d/postgresql stop');
+        if (fs.existsSync('/var/lib/postgresql/12/main')) {
+            execSync('rm -rfv /var/lib/postgresql/12/main/*');
+        }
+        execSync(`su -c "pg_basebackup -h ${remoteHostname} -U ${remoteOpDbUsername} -p 5432 -D /var/lib/postgresql/12/main/  -Fp -Xs -P -R" - postgres`);
+        execSync('/etc/init.d/postgresql start');
+        this.logger.trace('Postgres replication started successfully');
+    }
+
+    stopPostgresReplication() {
+        try {
+            this.logger.trace('Stopping postgres replication');
+            execSync('pg_ctlcluster 12 main promote');
+            this.logger.trace('Postgres replication stopped successfully');
+        } catch (error) {
+            if (error.message.includes('server is not in standby mode')) {
+                return;
+            }
+            throw error;
+        }
+    }
+
+    async isRemoteNodeAvailable() {
+        const {
+            remote_hostname,
+            is_remote_node_available_attempts_number,
+            is_remote_node_available_attempts_delay,
+            is_remote_node_available_attempts_timeout,
+        } = this.config.high_availability;
+
+        for (let i = 0; i < is_remote_node_available_attempts_number; i += 1) {
+            try {
+                // eslint-disable-next-line no-await-in-loop
+                const response = await this.otNodeClient.healthCheck(
+                    remote_hostname,
+                    is_remote_node_available_attempts_timeout,
+                    this.config.high_availability.active_node_data_sync_use_ssl,
+                );
+                if (response.statusCode === 200) {
+                    return true;
+                }
+                this.logger.trace('Remote node is not in active state. attempt: ', i + 1);
+            } catch (error) {
+                this.logger.trace(`Unable to fetch health check for remote node error: ${error.message}, attempt: ${i + 1}`);
+            }
+            if (i + 1 < is_remote_node_available_attempts_number) {
+                // eslint-disable-next-line no-await-in-loop
+                await this.sleepForMiliseconds(is_remote_node_available_attempts_delay);
```

Comment:
If we need this function in multiple places it would be great to put it in a single module

Reply:
Moved to Utilities